src/lib/ext-activitylog.js

```diff
@@ -147,15 +159,18 @@ class View {
   handleEvent(event) {
     if (event.type === 'click') {
       switch (event.target) {
-        case this.saveLogBtn:
-          this.saveLogBtn.dispatchEvent(new CustomEvent('savelog'));
-          break;
         case this.clearLogBtn:
           this.clearLogBtn.dispatchEvent(new CustomEvent('clearlog'));
           break;
-        default:
-          throw new Error(`wrong event target - ${event.target.tagName}`);
+        case this.optionsBtn:
+          this.optionsDropdown.hidden = !this.optionsDropdown.hidden;
+          break;
+        case this.saveLogBtn:
+          this.saveLogBtn.dispatchEvent(new CustomEvent('savelog'));
+          break;
       }
+
+      this.hideDropdowns(event.target);
```

Comment:
The `hideDropdowns` logic works but it requires a global click listener runs too often. In https://github.com/mozilla/extension-activity-monitor/pull/52#pullrequestreview-486879126 I said "consider taking advantage of the fact that only one dropdown is supposed to be visible at any time."

I was thinking of this:

```
class DropDownController {
  showDropDown(elem) {
    this.hideDropDown();
    this.elem = elem;
    document.addEventListener("click", this);
    // TODO: Actually show elem
  }
  hideDropDown() {
    let {elem} = this;
    if (!elem) {
      return;
    }
    this.elem = null;
    document.removeEventListener("click", this);
    // TODO: Actually hide elem
  }
}
```

This is much simpler than your current implementation where you need to keep track of all elements.

Reply:
Thanks for this. I have implemented like this.